plugin/storage/kafka/options.go

```diff
@@ -43,13 +43,17 @@ const (
 	suffixCompression      = ".compression"
 	suffixCompressionLevel = ".compression-level"
 	suffixProtocolVersion  = ".protocol-version"
+	suffixLingerMS         = ".linger-ms"
+	suffixBatchSize        = ".batch-size"
 
 	defaultBroker           = "127.0.0.1:9092"
 	defaultTopic            = "jaeger-spans"
 	defaultEncoding         = EncodingProto
 	defaultRequiredAcks     = "local"
 	defaultCompression      = "none"
 	defaultCompressionLevel = 0
+	defaultLingerMS         = 10
+	defaultBatchSize        = 1024
```

Comment:
Do these match the current defaults? We need to be backwards compatible.

Reply:
Here is Samara's code with comments.  I am not seeing default defined for `Frequency` and `Bytes`. I am not sure if it defaults to the Go type's default value.
```
		// The following config options control how often messages are batched up and
		// sent to the broker. By default, messages are sent as fast as possible, and
		// all messages received while the current batch is in-flight are placed
		// into the subsequent batch.
		Flush struct {
			// The best-effort number of bytes needed to trigger a flush. Use the
			// global sarama.MaxRequestSize to set a hard upper limit.
			Bytes int
			// The best-effort number of messages needed to trigger a flush. Use
			// `MaxMessages` to set a hard upper limit.
			Messages int
			// The best-effort frequency of flushes. Equivalent to
			// `queue.buffering.max.ms` setting of JVM producer.
			Frequency time.Duration
			// The maximum number of messages the producer will send in a single
			// broker request. Defaults to 0 for unlimited. Similar to
			// `queue.buffering.max.messages` in the JVM producer.
			MaxMessages int
		}
```

Reply:
The intended defaults are defined in the [Kafka documentation](https://kafka.apache.org/documentation/). Whether sarama used them is a different issue though - which should we align with, especially if maybe going to change Kafka client lib in the future?

Reply:
I was able to find the default Samara values by debugging into a standalone Samara client.  The default values under config.Producer.Flush are all 0.  ie. go type's default value.   @objectiser @yurishkuro Please let me know if I should set the defaults to 0 for backward compatibility or set the defaults according to Kafka doc.  I am fine either way because the defaults are not performant for a production system.  We will be setting our own values as we tune Jaeger/Kafka for performance.

Reply:
Normally I would say we need to be backward compatible, which is fine for `linger.ms` as its default in kafka docs is also zero. However if `batch.size` is 0, it means batching is disabled (i.e. "_A small batch size will make batching less common and may reduce throughput (a batch size of zero will disable batching entirely)_", which probably accounts for the bad performance.

@yurishkuro Should this be treated as a bug, and set the batch size to the official Kafka doc value `16384`?

Reply:
Per @yurishkuro's comments in another part.  I set default to 0.